### PR TITLE
[GDnative] Disable lto for gdnative wrapper

### LINKED ISSUE #14976: [GDnative] Disable lto for gdnative wrapper static lib
Given the very nature of the gdnative wrapper lib (i.e. being compiled by 3rd party to expose) it can be compiled with a wide range of compilers. However lto options seems to go in the way:

With gcc 5.4 the compilation errored with a `lto1: fatal error: bytecode stream generated with LTO version 6.0 instead of the expected 4.1`. This is not cool because most recent version of gcc are not widely available on distributions (especially on CI platforms such as travis).

With clang the compilation succeed but symbols exposed by the wrapper lib are stripped out:
```
$ nm libpythonscript.so | grep gdnative_wrapper_
nm: libpythonscript.so: plugin needed to handle lto object
                 U _gdnative_wrapper_api_struct
                 U _gdnative_wrapper_arvr_api_struct
                 U _gdnative_wrapper_nativescript_api_struct
                 U _gdnative_wrapper_pluginscript_api_struct
```

On the other hand I'm not sure lto brings any benefit here (given this static library is small and all symbols should be used anyway...). I'm far from being an expert in this area so I might be missing something here.

### PR DESCRIPTION
**Pull request by touilleMan**
Closes #14976

### CODE REVIEW

File: modules/gdnative/SCsub [hunk 1]
```diff
@@ -245,6 +245,14 @@ if ARGUMENTS.get('gdnative_wrapper', False):
     gd_wrapper_env = env.Clone()
     gd_wrapper_env.Append(CPPPATH=['#modules/gdnative/include/'])
 
+    if gd_wrapper_env['use_lto']:
+        if env.msvc:
```
Comment (akien-mga) [inline modules/gdnative/SCsub]: Did you test it? ;)

Comment (akien-mga) [reply modules/gdnative/SCsub]: (Just hinting that you got your logic backwards :))

Comment (touilleMan) [reply modules/gdnative/SCsub]: haha you're right I count on the CI to do the test for me on this one ;-)

...

Which tells me I should open an issue to enable gdnative wrapper compilation on CI don't you think ?

Comment (touilleMan) [reply modules/gdnative/SCsub]: aaaaaand push force on the PR to hide my mistake !

**Pull request merged**